extensions/objects/templates/unicode-string-editor.component.spec.ts

```diff
@@ -0,0 +1,201 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for unicode string editor.
+ */
+
+import { EventEmitter, SimpleChanges } from '@angular/core';
+import { ExternalSaveService } from 'services/external-save.service';
+import { FormsModule } from '@angular/forms';
+import { UnicodeStringEditorComponent } from './unicode-string-editor.component';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+describe('UnicodeStringEditorComponent', () => {
+  let component: UnicodeStringEditorComponent;
+  let fixture: ComponentFixture<UnicodeStringEditorComponent>;
+  let externalSaveService: ExternalSaveService;
+
+  var externalSaveEventEmitter = new EventEmitter();
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [UnicodeStringEditorComponent]
+    }).compileComponents();
+  }));
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UnicodeStringEditorComponent);
+    component = fixture.componentInstance;
+    externalSaveService = TestBed.get(ExternalSaveService);
+    spyOnProperty(externalSaveService, 'onExternalSave').and.returnValue(
+      externalSaveEventEmitter);
+    component.alwaysEditable = false;
+    component.value = 'random value';
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should initialize component when interaction rule is selected', () => {
+    spyOn(component, 'closeEditor').and.callThrough();
+    spyOn(component.componentSubscriptions, 'add');
+    component.active = true;
+
+    expect(component.active).toBe(true);
+
+    component.ngOnInit();
+
+    expect(component.componentSubscriptions.add).toHaveBeenCalled();
+    expect(externalSaveService.onExternalSave).toEqual(
+      externalSaveEventEmitter);
+    expect(component.componentSubscriptions.closed).toBe(false);
+    expect(component.alwaysEditable).toBe(false);
+    expect(component.closeEditor).toHaveBeenCalled();
+    expect(component.active).toBe(false);
+  });
+
+  it('should not add subscription when alwaysEditable is true', () => {
+    spyOn(component, 'closeEditor').and.callThrough();
+    spyOn(component.componentSubscriptions, 'add');
+    component.alwaysEditable = true;
+    component.active = true;
+
+    expect(component.active).toBe(true);
+
+    component.ngOnInit();
+
+    // We cannot test what componentSubscriptions.add has been called with since
+    // externalSaveService is a private variable and inside it is an anonymous
+    // function.
+    expect(component.componentSubscriptions.add).not.toHaveBeenCalled();
+    expect(component.alwaysEditable).toBe(true);
```

Comment:
Each of your post-checks should have corresponding pre-checks (to verify that the change was caused as a result of the action being taken in the test).